stablehlo/reference/Ops.cpp

```diff
@@ -439,6 +444,83 @@ Tensor evalCeilOp(const Tensor &operand, ShapedType resultType) {
   return result;
 }
 
+Tensor evalCholeskyOp(const Tensor &a, bool lower, ShapedType resultType) {
+  Tensor result(resultType);
+  auto aShape = a.getShape();
+  auto cholesky = [&lower](const Tensor &A) {
+    Tensor L(A.getType());
+    auto conjugate = [&](const Element &el) {
+      return Element(el.getType(),
+                     std::complex<APFloat>(el.getComplexValue().real(),
+                                           -el.getComplexValue().imag()));
+    };
+
+    for (auto it = L.index_begin(); it != L.index_end(); ++it)
+      L.set(*it, Element(A.getElementType(), 0.0));
+
+    for (auto i = 0; i < A.getShape()[0]; ++i) {
+      for (auto j = 0; j <= i; ++j) {
+        Element sum(A.getElementType(), 0.0);
+        for (auto k = 0; k < j; ++k) {
+          if (isSupportedComplexType(A.getElementType()))
+            sum = sum + L.get(Index({i, k})) * conjugate(L.get(Index({j, k})));
+          else
+            sum = sum + L.get(Index({i, k})) * L.get(Index({j, k}));
+        }
+        if (i == j)
+          L.set(Index({i, j}), sqrt(A.get(Index({i, i})) - sum));
+        else
+          L.set(Index({i, j}),
+                ((A.get(Index({i, j})) - sum) / L.get(Index({j, j}))));
+      }
+    }
+
+    if (lower) return L;
+
+    if (isSupportedComplexType(A.getElementType()))
+      for (auto it = L.index_begin(); it != L.index_end(); ++it)
+        L.set(*it, conjugate(L.get(*it)));
+    return evalTransposeOp(L, {1, 0}, L.getType());
+  };
+
+  if (a.getRank() == 2) return cholesky(a);
+
+  auto getScalarTensor = [&](auto value) {
+    return makeTensor(DenseElementsAttr::get(
+        RankedTensorType::get({},
+                              IntegerType::get(a.getType().getContext(), 64)),
+        {value}));
+  };
+
+  SmallVector<int64_t> aNonBatchShape(aShape.end() - 2, aShape.end());
```

Comment:
Can we make `aNonBatchShape` -> `nonBatchingSizes`? This was inspired from the very next statement. I believe they should be named in a way so that they seem related.